### PR TITLE
V0.4.3

### PR DESCRIPTION
**Pull request by NoLongerBreathedIn**
Mostly added a load of casts.

### CODE REVIEW

**Comment by CoffeeTableEspresso** [comment]
Looks like you've broken the automated tests with your changes:

https://travis-ci.org/yasl-lang/yasl/jobs/528233153

If you fix these up, then I can merge this in (and I'll add some tests to make sure all the C++ stuff is alright).

**Comment by NoLongerBreathedIn** [comment]
I'll try to figure this out, with a hefty helping of gdb.

**Comment by CoffeeTableEspresso** [comment]
I'd first look at the spots where you removed the flexible array members. I suspect that's what's causing the issue. It's fine if you leave them in as long as the C++ compiler supports them.

**Comment by CoffeeTableEspresso** [comment]
You certainly did not fix it. You just turned on debug mode so writing to any memory is fine. If you look at the automated tests (the red x on the PR page) you'll see lots of them are still failing.

**Comment by NoLongerBreathedIn** [comment]
That was a different error. Please wait a bit. The error I fixed was in the CMake file. I'm currently trying to figure it out. Also, those failures were because of a changed in the test code that caused failures to be reported when the code was correct.

I said 'fixed error'. Singular. I didn't say I'd fixed more than one.

**Comment by CoffeeTableEspresso** [comment]
Sorry, I'll let you do your thing.

**Comment by NoLongerBreathedIn** [comment]
Well, it now works properly.

**Comment by CoffeeTableEspresso** [comment]
thanks. It's looking good now. I'm just working on a few changes before I merge this in, mainly adding automated tests to check that it compiles as C++ (so this doesn't get broken in the future). Once I'm done with that I'll merge this in.

**Comment by NoLongerBreathedIn** [comment]
Turned out not to need gdb, just a search for where the parser was freeing nodes. Changed to call a special function that deleted the children list first. (For the errors in the compiler test, check test/test_compiler/yats.h.)

File: interpreter/VM.c [hunk 1]
```diff
@@ -580,7 +580,7 @@ int vm_run(struct VM *vm) {
 		case DCONST_0:
 		case DCONST_1:
 		case DCONST_2:
-			vm_pushfloat(vm, opcode - DCONST_0); // make sure no changes to opcodes ruin this
+			vm_pushfloat(vm, (double)(opcode - DCONST_0)); // make sure no changes to opcodes ruin this
```
Comment (CoffeeTableEspresso) [inline interpreter/VM.c]: this should be `yasl_float`.

**Comment by CoffeeTableEspresso** [comment]
How did you compile this? I'm having a bit of trouble getting CMake to compile this as a C++ program.

**Comment by CoffeeTableEspresso** [comment]
Great work, I've merged this in. I'll be adding some C++ tests to make sure it continues to compile as a C++ program in the future.

File: interpreter/VM.c (hunk 1 above)
Comment (NoLongerBreathedIn) [reply interpreter/VM.c]: Whoops. Feel free to change.

**Pull request merged**